source/async_stack.cpp

```diff
@@ -38,11 +41,46 @@ extern "C" {
 // Initialise to some value that will be interpreted as an invalid key.
 inline pthread_key_t folly_async_stack_root_tls_key = 0xFFFF'FFFFu;
 }
-
-#endif  // FOLLY_ASYNC_STACK_ROOT_USE_PTHREAD
+#endif  //UNIFEX_ASYNC_STACK_ROOT_USE_PTHREAD
 
 namespace unifex {
 
+#if UNIFEX_ASYNC_STACK_ROOT_USE_PTHREAD == 0
+#include <vector>
+
+struct AsyncStackRootHolderList {
+  std::vector<void*> asyncStackRootHolders_;
```

Comment:
Might want to have a reserve here of some reasonable number to avoid allocations when pushing new roots.